app/components/channel_loader/channel_loader.js

```diff
@@ -6,152 +6,56 @@ import PropTypes from 'prop-types';
 import {
     Platform,
     View,
-    Animated,
-    Easing,
 } from 'react-native';
-import LinearGradient from 'react-native-linear-gradient';
+import Placeholder from 'rn-placeholder';
 
 import {changeOpacity, makeStyleSheetFromTheme} from 'app/utils/theme';
 
-const GRADIENT_START_END = 0;
-const GRADIENT_MIDDLE = 0.8;
-
-const START_VALUE = 0;
-const MIDDLE_VALUE = 50;
-const END_VALUE = 100;
-const DURATION = 2000;
-
 export default class ChannelLoader extends PureComponent {
     static propTypes = {
+        backgroundColor: PropTypes.string,
         channelIsLoading: PropTypes.bool.isRequired,
-        deviceWidth: PropTypes.number.isRequired,
         theme: PropTypes.object.isRequired,
     };
 
-    constructor() {
-        super();
-        this.animation = new Animated.Value(0);
-    }
-
-    componentDidMount() {
-        if (this.props.channelIsLoading) {
-            this.start();
-        }
-    }
-
-    componentWillUnmount() {
-        this.animation.stopAnimation();
-    }
-
-    componentWillReceiveProps(nextProps) {
-        if (nextProps.channelIsLoading !== this.props.channelIsLoading) {
-            if (nextProps.channelIsLoading) {
-                this.start();
-            } else {
-                this.animation.stopAnimation();
-            }
-        }
-    }
-
-    start() {
-        this.animation.setValue(START_VALUE);
-        Animated.sequence([
-            Animated.timing(this.animation, {
-                toValue: END_VALUE,
-                duration: DURATION,
-                easing: Easing.linear,
```

Comment:
Did you by any chance try `useNativeDriver: true` here?

https://facebook.github.io/react-native/blog/2017/02/14/using-native-driver-for-animated.html

Reply:
yes I did, and the problem was not the animation but the time it took to compute everything else

Reply:
Oh ok Thanks. Just one minor thing. `fade` animation works great but we initially had something similar to `shine` as expected. Last time I checked with `shine` this library had a really poor animation sequence. Is that the reason why you changed to `fade` here?

Reply:
yes, shine animation is bad really bad